pkg/agent/controller/supportbundlecollection/support_bundle_controller_test.go

```diff
@@ -0,0 +1,141 @@
+	"golang.org/x/crypto/ssh"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+	"k8s.io/utils/exec"
+
+	"antrea.io/antrea/pkg/agent/config"
+	agentquerier "antrea.io/antrea/pkg/agent/querier"
+	cpv1b2 "antrea.io/antrea/pkg/apis/controlplane/v1beta2"
+	fakeversioned "antrea.io/antrea/pkg/client/clientset/versioned/fake"
+	"antrea.io/antrea/pkg/ovs/ovsctl"
+	"antrea.io/antrea/pkg/querier"
+	"antrea.io/antrea/pkg/support"
+)
+
+type fakeController struct {
+	*SupportBundleController
+	mockController *gomock.Controller
+}
+
+func newFakeController(t *testing.T) (*fakeController, *fakeversioned.Clientset) {
+	controller := gomock.NewController(t)
+	clientset := &fakeversioned.Clientset{}
+	supportBundleController := NewSupportBundleController("vm1", config.ExternalNode.String(), "vm-ns", nil, nil,
+		nil, nil, true, true)
+	return &fakeController{
+		SupportBundleController: supportBundleController,
+		mockController:          controller,
+	}, clientset
+}
+
+func TestSupportbundleCollectionAdd(t *testing.T) {
+	testcases := []struct {
+		name                    string
+		supportBundleCollection *cpv1b2.SupportBundleCollection
+	}{
+		{
+			name:                    "Add SupportBundleCollection",
+			supportBundleCollection: generateSuppportbundleCollection("sftp://10.220.175.92:22/root/supportbundle"),
+		},
+		{
+			name:                    "Add SupportBundleCollection without url prefix",
+			supportBundleCollection: generateSuppportbundleCollection("10.220.175.92:22/root/supportbundle"),
+		},
+	}
+	for _, tt := range testcases {
+		t.Run(tt.name, func(t *testing.T) {
+			newAgentDumper = func(fs afero.Fs, executor exec.Interface, ovsCtlClient ovsctl.OVSCtlClient, aq agentquerier.AgentQuerier, npq querier.AgentNetworkPolicyInfoQuerier, since string, v4Enabled, v6Enabled bool) support.AgentDumper {
+				return &mockAgentDumper{}
+			}
+			getUploader = func(protocol string) (uploader, error) {
+				return &testUploader{}, nil
+			}
+			defer func() {
+				newAgentDumper = support.NewAgentDumper
+				getUploader = getUploaderByProtocol
+			}()
+			c, _ := newFakeController(t)
+			assert.NoError(t, c.generateSupportBundle(tt.supportBundleCollection))
+		})
+	}
+}
+
+type testUploader struct {
+}
+
+func (uploader *testUploader) upload(address string, path string, config *ssh.ClientConfig, tarGzFile io.Reader) error {
+	klog.Info("Called test uploader")
+	return nil
+}
+
+func generateSuppportbundleCollection(url string) *cpv1b2.SupportBundleCollection {
+	return &cpv1b2.SupportBundleCollection{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "SupportBundle",
+		},
+
+		FileServer: cpv1b2.BundleFileServer{
+			URL: url,
+		},
+		Authentication: cpv1b2.BundleServerAuthConfiguration{
+			BasicAuthentication: &cpv1b2.BasicAuthentication{
+				Username: "AAA",
+				Password: "BBBCCC",
+			},
+		},
+	}
+}
+
+type mockAgentDumper struct {
+}
+
+func (d *mockAgentDumper) DumpFlows(basedir string) error {
+	return nil
```

Comment:
Think about the case that AgentDumper returns error in some cornor cases, and add test case to cover it?